br/pkg/lightning/restore/get_pre_info.go

```diff
@@ -684,6 +703,18 @@ outloop:
 			return 0.0, false, errors.Trace(err)
 		}
 		lastRow := parser.LastRow()
+		if rowCount == 0 {
+			lastRowLen := len(lastRow.Row)
+			extendColsMap := make(map[string]int)
+			for i, c := range extendCols {
+				extendColsMap[c] = lastRowLen + i
+			}
+			for i, col := range tableInfo.Columns {
+				if p, ok := extendColsMap[col.Name.O]; ok {
+					columnPermutation[i] = p
```

Comment:
`columnPermutation` related function is very complex 😵 maybe check it has enough length for `i`

Reply:
`len(columPermutation) == len(tableInfo.Columns) + 1`. See https://github.com/pingcap/tidb/blob/c646842f2926b38c96bf05350033cce53fbce4ba/br/pkg/lightning/restore/table_restore.go#L876